Fix CTRL C Scoping

jdcaporale opened the pull request:
Addresses Issue #51. 
 
Moves the global variable into a single translation unit (common_header.cpp) and then accesses it via methods. 
Initial testing of this method worked, but this code remains untest and can be reviewed but DO NOT MERGE until tested. 

Also, handles a TODO that was stuck behind this fix, which moves LCM Subscriber source code into a cpp.

jdcaporale committed:
Fix CTRL-C scoping, Issue 51

jdcaporale committed:
Move LCM Subscriber Implementation to cpp, ref PR #65

ShaneRozenLevy commented on include/kodlab_mjbots_sdk/common_header.h:
```diff
@@ -9,18 +19,25 @@
+ */
+bool CtrlCDetected();
+
+// Ctrl-C signal handling adapted from solo8 repo
```
Does this comment still belong here?

jdcaporale replied:
```suggestion
```

ShaneRozenLevy reviewed (approved):
LGTM. One comment seemed out of place.

jdcaporale committed:
Change ODRI Solo8 attribution comment

The CTRL-C signal handling was originally using Solo8's implementation.
Since then things have changed, but some of it is still there so the attribution,
is still there.

jdcaporale commented:
Test and ready.
